### PR TITLE
Minor refactor

### PR DESCRIPTION
**Pull request by thiolliere**
simple minor refactor

### CODE REVIEW

File: node/core/runtime-api/src/lib.rs [hunk 1]
```diff
@@ -302,33 +302,15 @@ where
 	let _timer = metrics.time_make_runtime_api_request();
 
 	macro_rules! query {
-		($req_variant:ident, $api_name:ident (), $sender:expr) => {{
+		($req_variant:ident, $api_name:ident ($($param:expr),*), $sender:expr) => {{
 			let sender = $sender;
 			let api = client.runtime_api();
-			let res = api.$api_name(&BlockId::Hash(relay_parent))
+			let res = api.$api_name(&BlockId::Hash(relay_parent), $( $param.clone() ),*)
```
Comment (KiChjang) [inline node/core/runtime-api/src/lib.rs:308]: Perhaps this makes more sense?

```suggestion
			let res = api.$api_name(&BlockId::Hash(relay_parent) $(, $param.clone() )*)
```

**Review by KiChjang — APPROVED** [review]
Small nit, otherwise looks good.

File: node/core/runtime-api/src/lib.rs [hunk 2]
```diff
@@ -302,33 +302,15 @@ where
...
-				.map_err(|e| RuntimeApiError::from(format!("{:?}", e)));
-			metrics.on_request(res.is_ok());
-			let _ = sender.send(res.clone());
-
-			if let Ok(res) = res {
-				Some(RequestResult::$req_variant(relay_parent, $($param),+, res.clone()))
-			} else {
-				None
-			}
+			res.ok().map(|res| RequestResult::$req_variant(relay_parent, $( $param, )* res.clone()))
```
Comment (bkchr) [inline node/core/runtime-api/src/lib.rs:313]:
```suggestion
			res.ok().map(|res| RequestResult::$req_variant(relay_parent, $( $param, )* res))
```

**Review by bkchr — APPROVED** [review]

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**